Mathlib/Topology/Algebra/Category/ProfiniteGrp/Limits.lean

```diff
@@ -0,0 +1,133 @@
+its quotients by open normal subgroups.
+
+## Main definitions
+
+* `QuotientOpenNormalSubgroup` : The functor that maps open normal subgroup of a profinite group
+  `P` to the quotient group of it (which is finite, as shown by previous lemmas).
+
+* `toLimitQuotientOpenNormalSubgroup` : The continuous homomorphism from a profinite group `P` to
+  projective limit of its quotients by open normal subgroups ordered by inclusion.
+
+* `ContinuousMulEquivLimitQuotientOpenNormalSubgroup` : The `toLimitQuotientOpenNormalSubgroup` is a
+  `ContinuousMulEquiv`
+
+## Main Statements
+
+* `OpenNormalSubgroupSubClopenNhdsOfOne` : For any open neighborhood of `1` there is an
+  open normal subgroup contained within it.
+
+-/
+
+universe u
+
+open CategoryTheory TopologicalGroup
+
+namespace ProfiniteGrp
+
+theorem exist_openNormalSubgroup_sub_open_nhd_of_one {G : Type*} [Group G] [TopologicalSpace G]
+    [TopologicalGroup G] [CompactSpace G] [TotallyDisconnectedSpace G] {U : Set G}
+    (UOpen : IsOpen U) (einU : 1 ∈ U) : ∃ H : OpenNormalSubgroup G, (H : Set G) ⊆ U := by
+  rcases ((Filter.HasBasis.mem_iff' ((nhds_basis_clopen (1 : G))) U ).mp <|
+    mem_nhds_iff.mpr (by use U)) with ⟨W, hW, h⟩
+  rcases exist_openNormalSubgroup_sub_clopen_nhd_of_one hW.2 hW.1 with ⟨H, hH⟩
+  exact ⟨H, fun _ a ↦ h (hH a)⟩
+
+section
+
+instance (P : ProfiniteGrp) : SmallCategory (OpenNormalSubgroup P) :=
+  Preorder.smallCategory (OpenNormalSubgroup ↑P.toProfinite.toTop)
+
+/-- The functor from `OpenNormalSubgroup P` to `FiniteGrp` sending `U` to `P ⧸ U`,
+where `P : ProfiniteGrp`. -/
+def QuotientOpenNormalSubgroup (P : ProfiniteGrp) :
+    OpenNormalSubgroup P ⥤ FiniteGrp := {
+    obj := fun H => FiniteGrp.of (P ⧸ H.toSubgroup)
+    map := fun fHK => QuotientGroup.map _ _ (.id _) (leOfHom fHK)
+    map_id _ := QuotientGroup.map_id _
+    map_comp f g := (QuotientGroup.map_comp_map
+      _ _ _ (.id _) (.id _) (leOfHom f) (leOfHom g)).symm }
+
+/-- The continuous homomorphism from a profinite group `P` to the projective limit of
+its quotients by open normal subgroups ordered by inclusion.-/
+def toLimitQuotientOpenNormalSubgroup (P : ProfiniteGrp.{u}) : P ⟶
+    limit (QuotientOpenNormalSubgroup P ⋙ forget₂ FiniteGrp ProfiniteGrp) where
+  toFun p := ⟨fun H => QuotientGroup.mk p, fun _ => rfl⟩
+  map_one' := Subtype.val_inj.mp rfl
+  map_mul' x y := Subtype.val_inj.mp rfl
+  continuous_toFun := continuous_induced_rng.mpr (continuous_pi fun H ↦ by
+    dsimp
+    apply Continuous.mk
+    intro s _
+    rw [← (Set.biUnion_preimage_singleton QuotientGroup.mk s)]
+    refine isOpen_iUnion (fun i ↦ isOpen_iUnion (fun _ ↦ ?_))
+    convert IsOpen.leftCoset H.toOpenSubgroup.isOpen' (Quotient.out i)
+    ext x
+    simp only [Set.mem_preimage, Set.mem_singleton_iff]
+    nth_rw 1 [← QuotientGroup.out_eq' i, eq_comm, QuotientGroup.eq]
+    exact Iff.symm (Set.mem_smul_set_iff_inv_smul_mem) )
+
+theorem toLimitQuotientOpenNormalSubgroup_dense (P : ProfiniteGrp.{u}) : Dense <|
```

Comment:
Why don't you use the `DenseRange` def? Also, I think that `dense` (or `denseRange`) should come first in the declaration name .